src/Uno.UI.Runtime.Skia.Android/AndroidSkiaTextBoxNotificationsProviderSingleton.cs

```diff
@@ -31,7 +32,15 @@ public void OnUnfocused(TextBox textBox)
 	{
 		if (UnoSKCanvasView.Instance is { } canvasView)
 		{
-			canvasView.TextInputPlugin.HideTextInput();
+			// Hide the keyboard only when the next focused element is not a TextBox
+			// This prevents the keyboard from flickering when switching between TextBoxes
+			// https://github.com/unoplatform/uno-private/issues/1160
+			var xamlRoot = textBox.XamlRoot;
+			if (xamlRoot is null || FocusManager.GetFocusedElement(xamlRoot) is not TextBox)
```

Comment:
we are also not checking for PasswordBox. while this is okay atm, since PBox inherits from TextBox, but it is not the case in winrt, and could fail when we eventually aligned with winrt.